app/controllers/saml_idp_controller.rb

```diff
@@ -156,11 +156,13 @@ def log_external_saml_auth_request
   end
 
   def requested_ial
-    requested_ial_acr = FederatedProtocols::Saml.new(saml_request).ial
-    requested_ial_component = Vot::AcrComponentValues.by_name[requested_ial_acr]
-    return 'ialmax' if requested_ial_component&.requirements&.include?(:ialmax)
-
-    saml_request&.requested_ial_authn_context || 'none'
+    saml_protocol = FederatedProtocols::Saml.new(saml_request)
+    requested_ial_acr = saml_protocol.ial
+    if requested_ial_acr == ::Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
+      return 'ialmax'
+    else
+      saml_protocol.requested_ial_authn_context.presence || 'none'
```

Comment:
this is not in scope to your change, but interesting that we return 'none' here instead of nil 🤔